File: 2학년 1학기/기타/곱셈.cpp
#include <iostream>

int GetMod(int base, int exp, int mod)
{
    if (exp == 0)
        return 1 % mod;
    else if (exp == 1)
        return base % mod;

    int mid = exp / 2;
    return (int)(((long long)GetMod(base, mid, mod) * GetMod(base, exp - mid, mod)) % mod);
}

int main()
{
    int a, e, m;
    scanf("%d %d %d", &a, &e, &m);
    printf("%d", GetMod(a, e, m));

    return 0;
}

File: 3학년 1학기/재귀함수 세미나/곱셈.py
def power(val, exp, mod):
    if exp == 0:
        return 1 % mod
    elif exp % 2 == 0:
        return (power(val, exp//2, mod) ** 2) % mod
    else:
        return ((power(val, exp//2, mod) ** 2) % mod * val) % mod

[a, b, c] = list(map(int, input().split(" ")))
print(power(a, b, c))